lockfile/src/golang.rs

```diff
@@ -66,4 +66,69 @@ mod tests {
             assert!(pkgs.contains(&expected_pkg));
         }
     }
+
+    #[cfg(feature = "generator")]
+    #[test]
+    fn lock_generate_goproj() {
+        let path = Path::new("../tests/fixtures/lock_generate_go/go.mod");
+        let lockfile = GoGenerator.generate_lockfile(path).unwrap();
+
+        let pkgs = GoSum.parse(&lockfile).unwrap();
+        assert_eq!(pkgs.len(), 9);
```

Comment:
Has not really be an issue in the past. Figuring it out is usually very straight-forward and logging all packages is easy.

Reply:
Your sorting doesn't actually fix this here.

Reply:
Then I'm not sure what the issue is.

Reply:
When asserting that the collections are equal, don't compare the lengths. Alternatively, if you're doing tests for whether the collection contains the expected elements, comparing the length last would give more useful output in the most likely failure case where something is missing. When the length assertion fails it just says the length is wrong and you have to step through with a debugger or modify the code in order to know why the lengths are different.

Reply:
Gotcha, I was just being consistent with [other tests](https://github.com/phylum-dev/cli/blob/c6e1f78deda3c02bef73d6a70252a8599430d01d/lockfile/src/csharp.rs#L166). No one seemed to have an issue there. There's a test further below that actually tests if they are equal, but I'll remove this length check.

Reply:
> No one seemed to have an issue there.

As I've said, this has never been an issue. I don't mind it at all personally.

That said, most other tests do this because comparing the entire vec is just unnecessary noise, so they compare the length to check for approximate correctness and then take a few edgecase samples to confirm specific values.